project-base/src/Shopsys/ShopBundle/Component/Flysystem/VolumeDriver.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ShopBundle\Component\Flysystem;
+
+use Barryvdh\elFinderFlysystemDriver\Driver;
+use elFinder;
+use Shopsys\ShopBundle\Component\Flysystem\Plugin\GetUrl;
+
+class VolumeDriver extends Driver
+{
+    protected function configure()
+    {
+        @parent::configure();
+
+        // set thumbnails path
+        $path = $this->options['tmbPath'];
+
+        if ($path) {
+            if (!$this->fs->has($path)) {
+                if ($this->_mkdir($path, '')) {
+                    $this->_chmod($path, $this->options['tmbPathMode']);
+                } else {
+                    $path = '';
+                }
+            }
+
+            $stat = $this->_stat($path);
+
+            if ($this->_dirExists($path) && $stat['read']) {
+                $this->tmbPath = $path;
+                $this->tmbPathWritable = $stat['write'];
+            }
+        }
+
+        $this->fs->addPlugin(new GetUrl($this->options));
+    }
+
+    /**
+     * @param string $hash
+     */
+    public function tmb($hash)
+    {
+        $path = $this->decode($hash);
+        $stat = $this->_stat($path, $hash);
+
+        if (isset($stat['tmb'])) {
+            $res = $stat['tmb'] == '1' ? $this->createTmb($path, $stat) : $stat['tmb'];
+
+            if (!$res) {
+                list($type) = explode('/', $stat['mime']);
+                $fallback = $this->options['resourcePath'] . DIRECTORY_SEPARATOR . strtolower($type) . '.png';
+                if (is_file($fallback)) {
+                    $res = $this->tmbname($stat);
+                    if (!$this->fs->put($fallback, $this->tmbPath . DIRECTORY_SEPARATOR . $res)) {
+                        $res = false;
+                    }
+                }
+            }
+            return $res;
+        }
+        return false;
+    }
+
+    /**
+     * @param string $path
+     * @param array $stat
+     */
+    protected function gettmb($path, $stat)
```

Comment:
In these you just use filesystem instead of methods such as copy() and so on right? Or is there something else that im Missing